security-demo/README.md

```diff
@@ -0,0 +1,263 @@
+All communications are blocked from unverified workloads.
+
+* Certificate rotation: Short-lived certificates are generated and rotation policies can be configured for every service communication.
+There is no need for custom agents and reliance on specific orchestrators for certificate rotation configuration and management.
+
+* Automated non-root CA certificate heirarchical deployments: Edge spire servers can be configured to not share any root CA chain for downstream nodes and workloads.
+
+## SPIRE-based Identity Management Reference Architecture
+
+Goal of this example is to achieve workload identity management for
+cloud and edge processes . A reference architecture is shown below. A
+list of things to-do is listed below in version information section.
+For further integration , an extension add-on will be developed
+integrated with kubeege for the same.
+
+![image](./doc/images/demo-arch.png)
+
+## Source Folders
+* release : Directory has configurations and scripts to be used for deployment of identity management infrastructure for Kubeedge.
+
+* app-agent-conf  : Configurations for spire agent interfacing with edge (event-bus) and user applications . Spire agent communicates to edge spire server.|
+
+* conf : Configurations for spire server (cloud and edge) and spire agent (agent communicating to cloud spire server).
+
+* certs : Spiffe helper communicates with spire agents and download certificates in this folder . Ghostunnel is loaded with these certificates . For edge to user interface application , spiffe helper configuration and ghostunnel certs are present in event-bus and user-app folders respectively.
+
+* app-binaries : All binaries for edge , cloud and test applications.
+
+* log : Logs for different processes run as part of the example.
+
+## Scripts
+* deploy-cloud.sh : Deploys cloud part of identity management infrastructure. Used to 
+                    
+                    1) Start cloud spire server.
+                           
+                    2) Registers cloud node and cloud agent.
+                           
+                    3) Registers and starts cloud hub (test cloud hub command simulator).
+
+* deploy-edge.sh  : Deploys edge part of identity management infrastructure.
+                           
+                    1) Registers edge node and edge agent communication with cloud spire
+                           server.
+                           
+                    2) Registers and starts edge spire server.
+                           
+                    3) Registers and starts edge agent for edge to user app communication interface.
+
+* start-spiffe-helper.sh  : Based on the configuration (helper.conf) , starts spiffe-helper communication for certificate download and rotation. Ghostunnel is run using spiffe-helper.
+
+* commands.sh  : Abstracts spire cli commands.
+ 
+## Prerequisites
```

Comment:
The following prerequisites are common, so there is no need to add them?

Reply:
Prerequisites are based on spire, script (while read problems in older version bash) and package downloading capability dependencies.